app/models/manager_refresh/inventory_collection/builder.rb

```diff
@@ -0,0 +1,273 @@
+module ManagerRefresh
+  class InventoryCollection
+    # @see test in /spec/models/manager_refresh/inventory_collection/builder_spec.rb
+    class Builder
+      class MissingModelClassError < StandardError; end
+
+      require_nested :CloudManager
+      require_nested :InfraManager
+      require_nested :AutomationManager
+      require_nested :NetworkManager
+      require_nested :StorageManager
+      require_nested :List
+
+      include ::ManagerRefresh::InventoryCollection::Builder::Shared
+
+      # Default options for builder
+      #   :adv_settings
+      #     - values from Advanced settings (doesn't overwrite values specified in code)
+      #     - @see method ManagerRefresh::Inventory::Persister.options()
+      #   :adv_settings_enabled
+      #     - enable/disable this properties
+      #   :auto_missing_parent
+      #     - auto assigns parent if this property is missing
+      #     - @see method add_parent_if_missing()
+      #   :auto_model_class
+      #     - tries to set model_class from persister class
+      #     - @see method auto_model_class
+      #   :auto_inventory_attributes
+      #     - auto creates inventory_object_attributes from target model_class setters
+      #     - attributes used in InventoryObject.add_attributes
+      #   :shared_properties
+      #     - any properties applied if missing (not explicitly specified)
+      #   :without_model_class
+      #     - if false and no model_class derived or specified, throws exception
+      def self.default_options
+        {
+          :adv_settings              => {},
+          :adv_settings_enabled      => true,
```

Comment:
What is the purpose of this property?  When would we want to set true/false the entire set of advanced settings?

Reply:
This interface is common for current and future providers, so I want to give a choice to somebody to turn it off for for example targeted_collection persister.

Maybe there should be discussion about adv_settings as a whole. As a low-level properties it has no ability to redefine anything in IC definitions, so you cannot set any important property through it, only add missing, which doesn't make sense to me

Is it really used somewhere?

Reply:
Yeah I think these settings aren't really used for the IC properties, they're a level higher like "should we use graph refresh" or "should we use the targeted persister".  Maybe we can drop this and work on a way of actually using settings to drive IC properties in a follow-up, that could actually be really helpful.

Reply:
as I mentioned below, adv_settings contains only subkey :inventory_collections - which I suggest to remove from interface (and from settings yaml?)

Reply:
I think we can remove ```:adv_settings_enabled``` for now, less options the better. (we can add it later, if there would be a usecase for it)